examples/c++/vector-api/example.cpp

```diff
@@ -0,0 +1,188 @@
+#include <iostream>
+#include <vector>
+#include <memory>
+#include <cassert>
+
+#include "icicle/runtime.h"
+#include "icicle/api/bn254.h"
+#include "icicle/utils/log.h"
+
+
+// SP: I undertstand this code is auto-generated, but I can't get scrip/gen_c_api.py to work. 
+
+extern "C" eIcicleError bn254_vector_product(
```

Comment:
I think it's better to use the template function in vec_ops.h rather than those C apis that are for FFI.